haystack/components/evaluators/document_map.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, List
+
+from haystack import Document, component
+
+
+@component
+class DocumentMeanAveragePrecision:
+    """
+    Evaluator that calculates the mean average precision of the retrieved documents.
```

Comment:
Let's add a line here explaining what's being calculated (we can use the text from the Notion page).